nixos/modules/services/backup/sanoid.nix

```diff
@@ -0,0 +1,163 @@
+{ config, stdenv, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sanoid;
+
+  percentType = (types.ints.between 0 100) // {
+    name = "Percent";
+  };
+
+  commonOptions = {
+    hourly = mkOption {
+      description = "Number of hourly snapshots";
+      type = types.nullOr types.ints.unsigned;
+      default = null;
+    };
+
+    daily = mkOption {
+      description = "Number of daily snapshots";
+      type = types.nullOr types.ints.unsigned;
+      default = null;
+    };
+
+    monthly = mkOption {
+      description = "Number of monthly snapshots";
+      type = types.nullOr types.ints.unsigned;
+      default = null;
+    };
+
+    yearly = mkOption {
+      description = "Number of yearly snapshots";
+      type = types.nullOr types.ints.unsigned;
+      default = null;
+    };
+
+    autoprune = mkOption {
+      description = "Whether to automatically prune old snapshots.";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+
+    autosnap = mkOption {
+      description = "Whether to automatically take snapshots.";
+      type = types.nullOr types.bool;
+      default = null;
+    };
+
+    extraConfig = mkOption {
+      description = "Extra configuration for this template/dataset";
+      type = types.lines;
+      default = "";
+    };
+  };
+
+  datasetOptions = commonOptions // {
+    useTemplate = mkOption {
+      description = "Names of the templates to use for this dataset";
+      type = types.nullOr (types.listOf (types.enum (attrNames cfg.templates)));
```

Comment:
For the manual you should override this options description to note that only values in `templates` can be used, because while building docs `templates` is just gonna be empty.

Reply:
Is this also one of those situations where changing `templates` will cause unintended manual rebuilds?

Reply:
Yeah probably

Reply:
This doesn't seem to cause manual rebuilds as far as I can tell, but I overrode the type description anyway so that it is more useful than `list of one of s`.